Vector2.inl causing compiler errors with MSVC in VS2022

Thanks for raising your issue here! 🙂

Before you submit the issue however, we'd like you to consider the follow points.

* We like to use the issue tracker for confirmed issues.
* If you're stuck with SFML, please use [the forum](https://en.sfml-dev.org/forums/index.php#c3) to get help.

----

## Subject of the issue
Was attempting to test out changes from the most recent master snapshot of the SFML 3.0 changes and was unable to compile my project. This project I had been using to test some of the 3.0 changes with the previous snapshot build, when I updated to the most recent one and did a clean rebuild I noticed the errors. Another user in the Discord server for SFML also reported the same compilation errors.

```
include\SFML\System\Vector2.inl(293,41): error C2370: 'sf::Vector2<T>::UnitX': redefinition; different storage class
include\SFML\System\Vector2.hpp(204): message : see declaration of 'sf::Vector2<T>::UnitX'
include\SFML\System\Vector2.inl(296,41): error C2370: 'sf::Vector2<T>::UnitY': redefinition; different storage class
include\SFML\System\Vector2.hpp(205): message : see declaration of 'sf::Vector2<T>::UnitY'
```

I tried building the SFML from source, then building against the output of that and still the same issues occurred.

## Your environment

Microsoft Windows 11
64 Bit SFML 3.0 (occurs with binaries from snapshot taken from master branch on 03/03/2022 & occurs when using binaries built from source)
Visual Studio 2022 (VC 17)

## Steps to reproduce

- Setup a project which uses SFML 3.0 within VS2022
- Set C++ language version of project to C++17
- Use the source shown below
- Build and the compiler errors should occur

```cpp
#include <SFML/Graphics.hpp>

int main()
{
    sf::RenderWindow window(sf::VideoMode(1280, 720), "Minimal, complete and verifiable example");
    window.setFramerateLimit(60);

    while (window.isOpen())
    {
        sf::Event event;
        while (window.pollEvent(event))
        {
            if (event.type == sf::Event::Closed)
                window.close();
        }

        window.clear();
        window.display();
    }
}
```

## Expected behavior

Compilation should complete without errors.

## Actual behavior

The following compiler errors are thrown

```
include\SFML\System\Vector2.inl(293,41): error C2370: 'sf::Vector2<T>::UnitX': redefinition; different storage class
include\SFML\System\Vector2.hpp(204): message : see declaration of 'sf::Vector2<T>::UnitX'
include\SFML\System\Vector2.inl(296,41): error C2370: 'sf::Vector2<T>::UnitY': redefinition; different storage class
include\SFML\System\Vector2.hpp(205): message : see declaration of 'sf::Vector2<T>::UnitY'
```

> Maybe they still need to be explicitly `static constexpr` at their definition site? @Bambo-Borris could you try if adding `static` in the .inl fixes the issue?

@Bromeon I tried this change in the .inl file but it yields the following compilation errors

```
include\SFML\System\Vector2.inl(291): error C2720: 'sf::Vector2<T>::UnitX': 'static ' storage-class specifier illegal on members
include\SFML\System\Vector2.inl(294): error C2720: 'sf::Vector2<T>::UnitY': 'static ' storage-class specifier illegal on members
```

--- a/cmake/CompilerWarnings.cmake
+++ b/cmake/CompilerWarnings.cmake
@@ -27,7 +27,7 @@ function(set_file_warnings)
         /w14905 # wide string literal cast to 'LPSTR'
         /w14906 # string literal cast to 'LPWSTR'
         /w14928 # illegal copy-initialization; more than one user-defined conversion has been implicitly applied
-        # /permissive- # standards conformance mode for MSVC compiler. Disabled until all out-of-the-box Windows SDKs can successfully build with it.
+        /permissive- # standards conformance mode
 
         # Disables, remove when appropriate
         /wd4996 # disable warnings about deprecated functions

--- a/include/SFML/System/Vector2.inl
+++ b/include/SFML/System/Vector2.inl
@@ -288,7 +288,7 @@ constexpr bool operator !=(const Vector2<T>& left, const Vector2<T>& right)
 ////////////////////////////////////////////////////////////
 
 template <typename T>
-constexpr Vector2<T> Vector2<T>::UnitX(static_cast<T>(1), static_cast<T>(0));
+const Vector2<T> Vector2<T>::UnitX(static_cast<T>(1), static_cast<T>(0));
 
 template <typename T>
-constexpr Vector2<T> Vector2<T>::UnitY(static_cast<T>(0), static_cast<T>(1));
+const Vector2<T> Vector2<T>::UnitY(static_cast<T>(0), static_cast<T>(1));

